### PR TITLE
chore: remove deprecated fields in NSTemplateTier

### PR DESCRIPTION
**Pull request by xcoulon**
Fixes https://issues.redhat.com/browse/CRT-502

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

File: wait/member.go [hunk 1]
```diff
@@ -177,14 +175,14 @@ func (a *MemberAwaitility) WaitUntilNSTemplateSetDeleted(name string) error {
...
+func (a *MemberAwaitility) WaitForNamespace(username, ref string) (*v1.Namespace, error) {
 	namespaceList := &v1.NamespaceList{}
 	err := wait.Poll(a.RetryInterval, a.Timeout, func() (done bool, err error) {
 		namespaceList = &v1.NamespaceList{}
 		labels := map[string]string{
-			"toolchain.dev.openshift.com/owner":       username,
-			"toolchain.dev.openshift.com/type":        typeName,
-			"toolchain.dev.openshift.com/templateref": strings.ToLower(fmt.Sprintf("%s-%s-%s", tier, typeName, revision)),
+			"toolchain.dev.openshift.com/owner": username,
+			// "toolchain.dev.openshift.com/type":        typeName,
```
Comment (alexeykazakov) [inline wait/member.go:184]: I guess we don’t need that comment?

Comment (xcoulon) [reply wait/member.go]: Indeed! Removed in 5145b57

File: go.mod [hunk 2]
```diff
@@ -60,4 +60,6 @@ replace (
 	k8s.io/kube-openapi => k8s.io/kube-openapi v0.0.0-20200204173128-addea2498afe // avoids case-insensitive import collision: "github.com/googleapis/gnostic/openapiv2" and "github.com/googleapis/gnostic/OpenAPIv2"
 )
 
+replace github.com/codeready-toolchain/api => github.com/xcoulon/api v0.0.0-20200526122759-6061f7c8bf85
```
Comment (tinakurian) [inline go.mod:63]: just a reminder - don't forget this :-)

File: tiers/checks.go [hunk 3]
```diff
@@ -133,7 +147,7 @@ func (a *advancedTierChecks) limitRangeByType(nsType string) namespaceObjectsChe
 type teamTierChecks struct {
 }
 
-func (a *teamTierChecks) GetNamespaceObjectChecks(nsType string) []namespaceObjectsCheck {
+func (a *teamTierChecks) GetNamespaceObjectChecks(templateRef string) []namespaceObjectsCheck {
```
Comment (MatousJobanek) [inline tiers/checks.go:150]: the parameter should be `nsType` and not `templateRef`
```suggestion
func (a *teamTierChecks) GetNamespaceObjectChecks(nsType string) []namespaceObjectsCheck {
```

File: wait/member.go [hunk 4]
```diff
@@ -177,14 +175,13 @@ func (a *MemberAwaitility) WaitUntilNSTemplateSetDeleted(name string) error {
...
 
 // WaitForNamespace waits until a namespace with the given owner (username), type, revision and tier labels exists
-func (a *MemberAwaitility) WaitForNamespace(username, typeName, revision, tier string) (*v1.Namespace, error) {
+func (a *MemberAwaitility) WaitForNamespace(username, ref string) (*v1.Namespace, error) {
 	namespaceList := &v1.NamespaceList{}
 	err := wait.Poll(a.RetryInterval, a.Timeout, func() (done bool, err error) {
 		namespaceList = &v1.NamespaceList{}
 		labels := map[string]string{
 			"toolchain.dev.openshift.com/owner":       username,
-			"toolchain.dev.openshift.com/type":        typeName,
```
Comment (MatousJobanek) [inline wait/member.go:186]: the `toolchain.dev.openshift.com/type` label should be still present. You can also add a check for the other labels:
* `toolchain.dev.openshift.com/tier`
* `toolchain.dev.openshift.com/provider`

Comment (xcoulon) [reply wait/member.go:186]: done in 4b9cc3d

**Comment by xcoulon** [comment]
> Nice simplification of some parts of the tets 👍
> 
> BTW, please, remove the file `1` https://github.com/codeready-toolchain/toolchain-e2e/blob/919548ce5075919e6e3c5e830f66cd34d6e701ab/1

done in https://github.com/codeready-toolchain/toolchain-e2e/pull/116/commits/a5ebe7daf1f99f2a982c3bdec1eb008adafb1d44

File: go.mod (hunk 2 above)
Comment (xcoulon) [reply go.mod:63]: done in f63607d

**Pull request merged**